src/main/groovy/edu/oregonstate/mist/beaverbus/BeaverBusResource.groovy

```diff
@@ -0,0 +1,91 @@
+package edu.oregonstate.mist.beaverbus
+
+import com.codahale.metrics.annotation.Timed
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+import groovy.transform.CompileStatic
+
+import javax.annotation.security.PermitAll
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+
+@Path("beaverbus")
+@Produces(MediaType.APPLICATION_JSON)
+@PermitAll
+@CompileStatic
+class BeaverBusResource extends Resource {
+    private final RideSystemsDAO rideSystemsDAO
+    private final ResourceMapper mapper = new ResourceMapper()
+    private BeaverBusUriBuilder uriBuilder
+
+    BeaverBusResource(RideSystemsDAO rideSystemsDAO, URI endpointUri) {
+        this.rideSystemsDAO = rideSystemsDAO
+        this.endpointUri = endpointUri
+        this.uriBuilder = new BeaverBusUriBuilder(endpointUri)
+    }
+
+    @Path("routes")
+    @GET
+    @Timed
+    Response getRoutes() {
+        def routes = rideSystemsDAO.getRoutesForMapWithScheduleWithEncodedLine()
+        def routeResources = routes.collect { mapper.mapRoute(it, uriBuilder) }
+        ok(new ResultObject(data: routeResources)).build()
+    }
+
+    @Path("routes/{id}")
+    @GET
+    @Timed
+    Response getSingleRoute(@PathParam("id") Integer id) {
+        // RideSystems does not supply any way to get just a single route,
+        // so we have to fetch them all and then filter out the one we want
+        def routes = rideSystemsDAO.getRoutesForMapWithScheduleWithEncodedLine()
+        def route = routes.find { it.RouteID == id }
+        if (route == null) {
```

Comment:
maybe just using `if (!route)` ?

Reply:
I prefer to be explicit here. `find` returns `null` if there is no match, so that's what I check for. `!route` would have different semantics if `RouteWithSchedule` implemented `asBoolean`.